src/Traits/FactoryTrait.php

```diff
@@ -249,6 +249,63 @@ public static function createFromFormat($format, $time, $tz = null): ChronosInte
         return $dt;
     }
 
+    /**
+     * Creates a ChronosInterface instance from an array of date and time values.
+     *
+     * The 'year', 'month' and 'day' values must all be set for a date. The time
+     * values all default to 0.
+     *
+     * The 'timezone' value can be any format supported by `\DateTimeZone`.
+     *
+     * Allowed values:
+     *  - year
+     *  - month
+     *  - day
+     *  - hour
+     *  - minute
+     *  - second
+     *  - microsecond
+     *  - meridian ('am' or 'pm')
+     *  - timezone
+     *
+     * @param (int|string)[] $values Array of date and time values.
+     * @return static
+     */
+    public static function createFromArray($values): ChronosInterface
+    {
+        $values += ['hour' => 0, 'minute' => 0, 'second' => 0, 'microsecond' => 0];
+
+        $formatted = '';
+        if (
+            isset($values['year'], $values['month'], $values['day']) &&
+            (
+                is_numeric($values['year']) &&
+                is_numeric($values['month']) &&
+                is_numeric($values['day'])
+            )
+        ) {
+            $formatted .= sprintf('%04d-%02d-%02d', $values['year'], $values['month'], $values['day']);
+        }
+
+        if (isset($values['meridian']) && (int)$values['hour'] === 12) {
+            $values['hour'] = 0;
+        }
+        if (isset($values['meridian'])) {
+            $values['hour'] = strtolower($values['meridian']) === 'am' ? $values['hour'] : $values['hour'] + 12;
+        }
+        $formatted .= sprintf(
+            '%s%02d:%02d:%02d.%06d',
+            empty($formatted) ? '' : ' ',
```

Comment:
This space exists when date set, move this space up to above `sprintf` of date.

Reply:
I think the original code did this to avoid extra spaces when there is only date.  I can change it if you want.

Reply:
I know it came from original code. It would be one less condition and lower code complexity as there is either _date time_ or _time_, so space better inserted after _date_ part, not conditionally before _time_ part.

Reply:
You're right.  Fixed!